pkg/scheduler/framework/runtime/framework.go

```diff
@@ -655,14 +655,16 @@ func (f *frameworkImpl) RunPreFilterPlugins(ctx context.Context, state *framewor
 	var result *framework.PreFilterResult
 	var pluginsWithNodes []string
 	logger := klog.FromContext(ctx)
-	logger = klog.LoggerWithName(logger, "PreFilter")
-	// TODO(knelasevero): Remove duplicated keys from log entry calls
-	// When contextualized logging hits GA
-	// https://github.com/kubernetes/kubernetes/issues/111672
-	logger = klog.LoggerWithValues(logger, "pod", klog.KObj(pod))
+	verboseLogs := logger.V(4).Enabled()
+	if verboseLogs {
+		logger = klog.LoggerWithName(logger, "PreFilter")
+	}
 	for _, pl := range f.preFilterPlugins {
-		logger := klog.LoggerWithName(logger, pl.Name())
-		ctx := klog.NewContext(ctx, logger)
+		ctx := ctx
```

Comment:
This is a common pattern in this PR: loop iterations must not change the variable outside of the loop, so local variables get created if (and only if) needed.

When verbose logging is off, this is a cheap copy where the variable then doesn't get modified. When it's on, it gets modified.